_source/_posts/2022-05-13-express-prisma-and-tailwindcss.md

```diff
@@ -0,0 +1,665 @@
+
+## Create your Okta application
+
+Create a new directory for your application.  
+
+{% include setup/cli.md type="web" loginRedirectUri="http://localhost:3000/authorization-code/callback" %}
+
+## Create your Express Application
+
+Next, you will use the `express-generator` application generator tool to quickly scaffold your application.  Run the following command.
+
+```console
+npx express-generator --view=pug
+```
+
+Next, you will need to install your packages.  
+
+```console
+npm i @prisma/client@3.13.0
+npm i dotenv@16.0.0
+npm i passport@0.5.2
+npm i passport-openidconnect@0.1.1
+npm i express-session@1.17
+npm i -D tailwindcss@3.0.24
+npm i -D prisma@3.13.0
+```
+
+> `@prisma/client` is used to access the database from your server code
+> `dotenv` reads configuration settings from `.env` files like the one produced by the Okta CLI
+> `passport` is a middleware for Node.js that is flexible enough to handle most authentication scenarios, including Okta.  `passport-openidconnect` is a module for passport that lets you authenticate with OpenID Connect
+> `express-session` is required for passport.
+> `tailwindcss` is the CSS framework you will use.  If you've never used tailwindcss before you may wonder why it's a development dependency.  This is because tailwindcss will dynamically build your CSS files from your views and configuration.  More on that later
+> `prisma` is the ORM you are using.  This is a dev dependency because the prisma library handles the migrations, seed data, etc, while the `@prisma/client` package is used in your application at runtime.
+
+At this point, you should initialize tailwindcss and prisma even though you won't configure to use either quite yet.
+
+Start with prisma.  Run the following command.
+
+```console
+npx prisma init
+```
+
+This command will add a new folder named `prisma` to your application. This also adds a file called `.env` to your project with some default configuration. Replace the contents of that file with the code below.
+
+```bash
+# Environment variables declared in this file are automatically made available to Prisma.
+# See the documentation for more detail: https://pris.ly/d/prisma-schema#accessing-environment-variables-from-the-schema
+
+# Prisma supports the native connection string format for PostgreSQL, MySQL, SQLite, SQL Server, MongoDB, and CockroachDB (Preview).
+# See the documentation for all the connection string options: https://pris.ly/d/connection-strings
+
+DATABASE_URL={yourServersConnectionString}
+SEED_USER_NAME={yourOktaUserName}
+```
+
+In this example I used Microsoft's SQL Server because that's what I had installed at the time of writing this application.  But one of the cool things about Prisma and most ORMs is that they are built to support many databases.  The boilerplate code includes comments about how to configure your application for other database servers.  Feel free to use what you feel most comfortable with and use the appropriate connection string.
```

Comment:
I think installing and creating an empty SQLite DB is quicker to explain in a paragraph here.

Reply:
Easier for us to potentially check in the DB into the repo as well so that the demo can be run with minimal build steps.

Reply:
https://github.com/oktadev/okta-blog/pull/1172/commits/6a7fe3c83cf50cda128b7cc8326f539139c5fc77